ui/src/components/shared/UserDisplay.tsx

```diff
@@ -1,18 +1,112 @@
+import { Avatar, Card, Dropdown, Flex, Tag, Tooltip, Typography, theme } from "antd";
+import { useNavigate } from "react-router-dom";
+import { formatIdentifier } from "../../utils/forms";
+import IconCheck from "src/assets/icons/check.svg?react";
+import IconChevron from "src/assets/icons/chevron-selector-vertical.svg?react";
+import IconPlus from "src/assets/icons/plus.svg?react";
 import { useEnvContext } from "src/contexts/Env";
+import { useIdentityContext } from "src/contexts/Identity";
+import { ROUTES } from "src/routes";
+import { isAsyncTaskDataAvailable } from "src/utils/async";
+import { IDENTITY_ADD } from "src/utils/constants";
 
 export function UserDisplay() {
   const { issuer } = useEnvContext();
+  const { handleChange, identifier, identitiesList, identityDisplayName } = useIdentityContext();
+  const navigate = useNavigate();
+  const { token } = theme.useToken();
+
+  const identityItems = isAsyncTaskDataAvailable(identitiesList)
+    ? identitiesList.data.map((identity, index) => {
+        const currentIdentity = identity.identifier === identifier;
+        return {
+          key: `${index}`,
+          label: (
+            <Flex
+              align="center"
+              className={currentIdentity ? "active" : ""}
+              gap={16}
+              justify="space-between"
+            >
+              <Flex
+                style={{
+                  width: 242,
+                }}
+                vertical
+              >
+                <Tooltip title={identity.displayName}>
+                  <Typography.Text ellipsis>{identity.displayName}</Typography.Text>
+                </Tooltip>
+              </Flex>
+              {currentIdentity && <IconCheck />}
+            </Flex>
+          ),
+          onClick: () => handleChange(identity.identifier),
+        };
+      })
+    : [];
+
+  const items = [
+    {
+      key: "add",
+      label: (
+        <Flex gap={16}>
+          <IconPlus style={{ height: 20, width: 20 }} />
+          <Typography.Text>{IDENTITY_ADD}</Typography.Text>
+        </Flex>
+      ),
+      onClick: () => navigate(ROUTES.createIdentity.path),
+    },
+    ...identityItems,
+  ];
 
   return (
-    <Space>
-      <Avatar shape="square" size="large" src={issuer.logo} />
+    <Dropdown
+      menu={{ items }}
+      overlayClassName="identities-dropdown"
+      placement="bottom"
+      trigger={["click"]}
+    >
+      <Card className="user-display" styles={{ body: { cursor: "pointer", padding: 12 } }}>
+        <Flex gap={12} vertical>
+          <Flex justify="space-between">
+            <Avatar shape="square" size="large" src={issuer.logo} />
+            <IconChevron />
+          </Flex>
+          <Flex gap={4} vertical>
+            <Typography.Text
+              ellipsis={{ tooltip: identityDisplayName }}
+              style={{ fontWeight: 600 }}
+              type="secondary"
+            >
+              {identityDisplayName}
```

Comment:
What if there's no displayName? Show shortened DID instead. This is not very useful:
<img width="348" alt="image" src="https://github.com/user-attachments/assets/43e9943f-340d-40e7-85b8-c294b7654724">